Refresh Fonzie configuration

jbpenrath opened the pull request:
## Purpose
Currently pylint version installed is not compatible with python 2.7 so `make quality` failed.
Furthermore, the configuration of this project is not aligned with our other repositories, so it should be interesting to update some configuration. Finally Circle CI configuration seems to be wrong.

## Proposal
- [x] Pin all dependencies to avoid incompatibility issues
- [x] Install bandit avoid common security issues
- [x] Split lint commands to be align with our other repositories practice
- [x] Make Circle works again
- [x] Update Github templates

jbpenrath committed:
📌(all) pin all dependencies

As this project used Python 2.7, dependency compatiblity issue can be raised
easily. So we  pinned all dependencies to a working version to avoid this kind
issue for newcomers.

jbpenrath committed:
🚨(linter) update linter config

update linter config to be in aligned with our other repositories

jbpenrath committed:
🚨(bandit) add bandit linter

Add bandit linter to avoid common security issues.

sampaccoud commented on Makefile:
```diff
@@ -82,16 +82,46 @@ migrate: run ## run project migrations
 
 quality: ## check coding style with pycodestyle and pylint
-	touch tests/__init__.py
+quality: \
```
in other projects, this target is called "lint" or "lint-back"...

jbpenrath committed:
📝(github) update github templates

Use same github templates than our other repositories

jbpenrath committed:
💚(circleci) stop pulling docker image anonymously

Docker will start a rate limit on pulling images on docker hub. This rate
limit is based on the IP used for anonymous users, this is the CircleCI IP
that will be used so the rate limit will be reached quickly. If the image pull
is made with a connected user, the rate limit becomes a per-user rate so we
should not reach it once connected.

jbpenrath commented:
@sampaccoud I also update Circle CI configuration as some jobs failed without reason.
Also copy/paste github templates to be aligned with our other repositories.
